tests/FileUploadsTest.php

```diff
@@ -472,6 +472,73 @@ public function removing_first_item_from_array_of_temporary_uploaded_files_seria
         $component->call('removePhoto', 0);
         $this->assertStringStartsWith('livewire-files:', $component->get('photos'));
     }
+
+    /** @test */
+    public function removing_first_item_from_array_of_temporary_uploaded_files_serializes_correctly_with_in_array_public_property()
+    {
+        $file1 = UploadedFile::fake()->image('avatar1.jpg');
+        $file2 = UploadedFile::fake()->image('avatar2.jpg');
+        $file3 = UploadedFile::fake()->image('avatar3.jpg');
+        $file4 = UploadedFile::fake()->image('avatar4.jpg');
+
+        $component = Livewire::test(FileUploadInArrayComponent::class)
+                             ->set('obj.file_uploads', [$file1, $file2, $file3, $file4])
```

Comment:
AFAIK `set` is not setting `obj.file_uploads` as a nested array it's setting `obj.file_uploads` as a string, which would mean the test isn't complete even though it passes.

Reply:
i think it does setting `obj.file_uploads` as nested array and not as a string.
try dumping `$component->get('obj')` and `$component->get('obj.file_uploads')`

but i got Undefined index: obj.file_uploads in `$component->viewData('obj.file_uploads')` and a tweak was to get the temp files like this `$component->viewData('obj')['file_uploads']`